CRM/Contribute/Form/ContributionView.php

```diff
@@ -217,6 +217,13 @@ public function preProcess() {
       NULL,
       $recentOther
     );
+
+    $participantID = CRM_Core_DAO::getFieldValue('CRM_Event_DAO_ParticipantPayment',
+      $values['id'], 'participant_id', 'contribution_id'
+    );
+    $this->assign('contactId', $values['contact_id']);
+    $this->assign('hasPayment', $values['id']);
```

Comment:
A variable named hasPayment should use a boolean like isset($values['id']) rather than a numeric or NULL.

Just because a contribution is returned, I don't think that is equivalent to there being a payment already. Could you elaborate with a comment somewhere the reasons why this would be so?